src/app/modules/item/components/item-children-edit/item-children-edit.component.ts

```diff
@@ -1,22 +1,52 @@
 import { Component, Input, OnChanges } from '@angular/core';
 import { ItemChild } from './item-children';
+import { ItemData } from '../../services/item-datasource.service';
+import { GetItemChildrenService } from '../../http-services/get-item-children.service';
+import { Subscription } from 'rxjs';
 
 @Component({
   selector: 'alg-item-children-edit',
   templateUrl: './item-children-edit.component.html',
   styleUrls: [ './item-children-edit.component.scss' ]
 })
 export class ItemChildrenEditComponent implements OnChanges {
+  @Input() itemData?: ItemData;
 
-  @Input() data: ItemChild[] = [
-    { id: '100', title: 'First item', order: 1 },
-    { id: '50', title: 'Second item', order: 2 },
-  ];
+  state: 'loading' | 'error' | 'empty' | 'ready' = 'ready';
```

Comment:
Why hiding completely the component if it is empty ? The user still need to know there is no children to know he can add some.

Reply:
I've added a message when the chapter don't have child